File: TypeScript/GetStringLength.ts
// p76
const GetStringLength =async()=>{
    let len:number = 0
    let data:Array<string>=[]

    const readInterface = require(`readline/promises`).createInterface({
        input: process.stdin,
        output: process.stdout
    })

    console.log('文字列を入力してください')
    data = [...(await readInterface.question('>'))]

    while(typeof(data[len])!=="undefined"){
        len ++
    }

    readInterface.close()
    return len
}

(async()=>{
    const result = await GetStringLength()
    console.log(`文字長 = ${result}`)
})();


// const otherMethod =async()=>{
//     console.log('文字列を入力してください')
//     const data:string = await readInterface.question('>')

    
//     console.log(`文字長 = ${data.length}`)
//     readInterface.close()

//     return
// }

// otherMethod()

File: Python/get_string_length.py
# p76
def get_string_length() -> int:
    len = 0

    data = list(input('文字列を入力してください\n>'))

    for data[len] in data:
        len +=1

    return len

result = get_string_length()
print('文字長 = %d' % result )

# def other_method():
#     data = input('文字列を入力してください\n>')
#     print('文字長 = %s' % len(data) )

# other_method()